.github/workflows/scripts/check_commits.py

```diff
@@ -93,23 +93,45 @@ def check_commit(commit_sha):
     return ok
 
 
-def main():
-    logging.basicConfig(level=logging.INFO, format="%(levelname)s: %(message)s")
+def validate_push_commits(start_commit, end_commit):
+    commit_list = git_list_commits([start_commit, end_commit])
+    all_commits_ok = True
+    for commit_sha in commit_list:
+        LOG.info(f"Checking commit {commit_sha[:8]} ...")
+        if not check_commit(commit_sha):
+            all_commits_ok = False
+            break
+    return all_commits_ok
 
-    # pull_request = os.environ["TRAVIS_PULL_REQUEST"]
-    # repo_slug = os.environ["TRAVIS_REPO_SLUG"]
-    commit_range = os.environ["TRAVIS_COMMIT_RANGE"].split("...")
-    commit_list = git_list_commits(commit_range)
 
-    ok = True
+def validate_pr_commits(github_pr_commits_url):
+    request = requests.get(github_pr_commits_url)
+    commit_list = [c['sha'] for c in request.json()]
+
+    at_least_one_commit_ok = False
     for commit_sha in commit_list:
-        LOG.debug(f"Checking commit {commit_sha[:8]} ...")
-        if not check_commit(commit_sha):
-            ok = False
+        LOG.info(f"Checking commit {commit_sha[:8]} ...")
```

Comment:
Not sure what the errors look like when this fails, but I would like to see the actual commit message that failed shown.
(One less thing to dereference). 

If it's not shown elsewhere, may need to log it (I guess from check_commit()).